File: haskell/fizzbuzz_1000.hs
fizzbuzz :: Int -> String
fizzbuzz 1 = "1"
fizzbuzz n | (mod n 15) == 0 = (fizzbuzz(n-1)) ++ " " ++ "fizzbuzz"
           | (mod n 5)  == 0 = (fizzbuzz(n-1)) ++ " " ++ "buzz"
           | (mod n 3)  == 0 = (fizzbuzz(n-1)) ++ " " ++ "fizz"
           | otherwise       = (fizzbuzz(n-1)) ++ " " ++ show n


main = print $ fizzbuzz(1000)

File: cpp/fizzbuzz/fizzbuzz_1000.cpp
#include <iostream>
#include <string>

std::string fizzbuzz(int n) {
  if (n == 1)
    return std::to_string(n);
  else if (n % 15 == 0)
    return fizzbuzz(n-1) + " " + "fizzbuzz";
  else if (n % 5 == 0)
    return fizzbuzz(n-1) + " " + "buzz";
  else if (n % 3 == 0)
    return fizzbuzz(n-1) + " " + "fizz";
  else
    return fizzbuzz(n-1) + " " + std::to_string(n);
}

int main() {
  std::cout << fizzbuzz(1000);
}
